c/parallel/src/jit_templates/templates/operation.h

```diff
@@ -40,10 +81,26 @@ template <typename Tag, cccl_op_t_mapping Operation, cccl_type_info_mapping RetT
 struct stateful_user_operation
 {
   user_operation_state<Operation.size, Operation.alignment> state;
-  __device__ decltype(RetT)::Type operator()(decltype(ArgTs)::Type... args)
+  __device__ typename decltype(RetT)::Type operator()(typename decltype(ArgTs)::Type... args)
   {
-    return reinterpret_cast<decltype(RetT)::Type (*)(void*, decltype(ArgTs)::Type...)>(
-      Operation.operation)(&state, std::move(args)...);
+    // Note: The user provided C function (Operation.operation) must match the signature:
+    // void (void* state, void* arg1, ..., void* argN, void* result_ptr)
+    static constexpr cuda::std::size_t num_args = sizeof...(ArgTs);
```

Comment:
```suggestion
    constexpr cuda::std::size_t num_args = sizeof...(ArgTs);
```

There is no need for `static` here, since `constexpr` is defined in the function scope.